File: backend/src/main/java/com/backend/domain/repository/dto/response/RepositoryResponse.java
package com.backend.domain.repository.dto.response;

import com.backend.domain.repository.entity.Repositories;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

/**
 * Repository 기본 정보 응답 DTO
 * 사용자의 Repository 목록 조회 시 사용
 */
public record RepositoryResponse(
        Long id,
        String name,
        String description,
        String htmlUrl,
        boolean publicRepository,
        String mainBranch,
        List<String> languages,
        LocalDateTime createDate,
        Long ownerId
) {
    public RepositoryResponse(Repositories repositories) {
        this(
                repositories.getId(),
                repositories.getName(),
                repositories.getDescription(),
                repositories.getHtmlUrl(),
                repositories.isPublicRepository(),
                repositories.getMainBranch(),
                repositories.getLanguages().stream()
                        .map(lang -> lang.getLanguage().name())
                        .collect(Collectors.toList()),
                repositories.getCreateDate(),
                repositories.getUser().getId()
        );
    }
}

File: backend/src/main/java/com/backend/domain/repository/dto/response/RepositoryResponse.kt
package com.backend.domain.repository.dto.response

import com.backend.domain.repository.entity.Repositories
import java.time.LocalDateTime

/**
 * Repository 기본 정보 응답 DTO
 * 사용자의 Repository 목록 조회 시 사용
 */
data class RepositoryResponse(
    val id: Long,
    val name: String,
    val description: String?,
    val htmlUrl: String,
    val publicRepository: Boolean,
    val mainBranch: String,
    val languages: List<String>,
    val createDate: LocalDateTime,
    val ownerId: Long
) {
    companion object {
        @JvmStatic
        fun from(entity: Repositories): RepositoryResponse {

            val languages = entity.getLanguages()
                .map { it.language.name }

            return RepositoryResponse(
                id = entity.id!!,
                name = entity.name,
                description = entity.description,
                htmlUrl = entity.htmlUrl,
                publicRepository = entity.publicRepository,
                mainBranch = entity.mainBranch,
                languages = languages,
                createDate = entity.createDate!!,
                ownerId = entity.user?.id!!
            )
        }
    }
}
